deltacat/tests/compute/compact_partition_test_cases.py

```diff
@@ -0,0 +1,450 @@
+            pa.array([0.1] * 4 + [0.2] * 4 + [0.3] * 4 + [0.4] * 4 + [0.5] * 4),
+            pa.array(reversed([i for i in range(20)])),
+        ],
+        input_deltas_delta_type=DeltaType.UPSERT,
+        expected_terminal_compact_partition_result=pa.Table.from_arrays(
+            [
+                pa.array([0.1, 0.2, 0.3, 0.4, 0.5]),
+                pa.array([19, 15, 11, 7, 3]),
+            ],
+            names=["pk_col_1", "sk_col_1"],
+        ),
+        validation_callback_func=None,
+        validation_callback_func_kwargs=None,
+        create_placement_group_param=True,
+        records_per_compacted_file_param=DEFAULT_MAX_RECORDS_PER_FILE,
+        hash_bucket_count_param=None,
+        create_table_strategy=create_src_w_deltas_destination_strategy,
+    ),
+}
+
+REBASE_THEN_INCREMENTAL_TEST_CASES = {
+    "10-rebase-then-incremental-sanity": RebaseThenIncrementalCompactorTestCase(
+        primary_keys_param={"pk_col_1"},
+        sort_keys_param=[
+            {
+                "key_name": "sk_col_1",
+            },
+            {
+                "key_name": "sk_col_2",
+            },
+        ],
+        partition_keys_param=[{"key_name": "region_id", "key_type": "int"}],
+        partition_values_param=["1"],
+        column_names_param=["pk_col_1", "sk_col_1", "sk_col_2", "col_1"],
+        input_deltas_arrow_arrays_param=[
+            pa.array([str(i) for i in range(10)]),
+            pa.array([i for i in range(0, 10)]),
+            pa.array(["foo"] * 10),
+            pa.array([i / 10 for i in range(10, 20)]),
+        ],
+        input_deltas_delta_type=DeltaType.UPSERT,
+        rebase_expected_compact_partition_result=pa.Table.from_arrays(
+            [
+                pa.array([str(i) for i in range(10)]),
+                pa.array([i for i in range(0, 10)]),
+                pa.array(["foo"] * 10),
+                pa.array([i / 10 for i in range(10, 20)]),
+            ],
+            names=["pk_col_1", "sk_col_1", "sk_col_2", "col_1"],
+        ),
+        incremental_deltas_arrow_arrays_param=[
+            pa.array([str(i) for i in range(10)]),
+            pa.array([i for i in range(20, 30)]),
+            pa.array(["foo"] * 10),
+            pa.array([i / 10 for i in range(40, 50)]),
+        ],
+        incremental_deltas_delta_type=DeltaType.UPSERT,
+        expected_terminal_compact_partition_result=pa.Table.from_arrays(
+            [
+                pa.array([str(i) for i in range(10)]),
+                pa.array([i for i in range(20, 30)]),
+                pa.array(["foo"] * 10),
+                pa.array([i / 10 for i in range(40, 50)]),
+            ],
+            names=["pk_col_1", "sk_col_1", "sk_col_2", "col_1"],
+        ),
+        validation_callback_func=None,
+        validation_callback_func_kwargs=None,
+        create_placement_group_param=True,
+        records_per_compacted_file_param=DEFAULT_MAX_RECORDS_PER_FILE,
+        hash_bucket_count_param=None,
```

Comment:
Can you also add a test case with bigger hash bucket count than records? That way we will know the behavior if some hash buckets don't get utilized.

Reply:
Done